### DISCUSSION TITLE
Infinite Maze!

### DISCUSSION THREAD

**Discussion by GeekyGhost** [v8hid/infinite-zoom-automatic1111-webui#73] [Ideas]
So, I've been stitching videos together, reversing them, and all that. I just had an idea. What if... you had an infinite maze type of thing. Like everything works like it currently does right. There's just a pipeline now. Let me explain better. 

So, I take the starting frames and final frames from the videos and use them in animator and prompt shift with cnet, things like that. I take the last frame of that and use it with infinite zoom, so that I can create these continuous transitions and etc. 

I feel like a lot of this could be automated. A zoom in and out, have a setting for like an animator type of effect to take the final image of the zoom and "animate" it, or zoom back in, or other changes. Like you use a block system I think like the coding toy lol. Zoom in block, Zoom out block, Animate block for like an animator video generation type of effect. You could likely even incorporate film or modelscope! Sorry, just an idea. This is basically what I do manually, but, I think it could be automated to make some cool effects. You can even extract rhythm patterns and onsets and etc from audio to control some of the affects and transition so it's a unique tool, with a unique system, that's easy to use. I've worked on a concatenate with Rhythm function for my basic video that I'm still sorting some issues on, it's a challenge, but some of the results are really cool. That's probably a bit much to throw in at the moment with AI lol. 

It just seemed worth mentioning how I use the took and my thoughts on how it could be automated. Sorry! I love the Extension!! 😅👻

**Comment by v8hid** [comment]
Hey @GeekyGhost thank you so much for your feedback, I really liked your idea especially extracting rhythm patter from an audio and incorporate it with the interpolation.
I would like to hear more about your ideas and see the results you mentioned on animation, feel free to join our discord server for discussing more about it.

**Comment by GeorgLegato** [comment]
when we can control more than Zoom in and out, like left, right up and down, we can do some dungeon crawler thing.
That would imply a story-language like
"wwwassdweeeqq"
put your fingers to "awsd" as in 3d-.games and you understand the syntax. 
w: ZoomIN for one inpaintstep
s: ZoomOUT
a: Left
d: Right
e: Up
q: down

the example with 13 chars is then the amount of "Total inpainting steps", replacing the slider.


Let us do this, but first @v8hid needs to finish the rework the painting routine, so that we can compose/recompse movements in all directions.

**Reply by v8hid** [reply]
@GeorgLegato Such a cool idea, I will finish the rework as fast I can, it will be finished by tomorrow morning.

**Reply by v8hid** [reply]
We can also add spin to this, e.g., `s>` means zooming out clockwise

**Reply by GeekyGhost** [reply]
My initial idea was like a big canvas and have the box be the mask and have it move over the canvas lol. This way all you're really doing is something similar to open outpaint with everything built in and you're just moving the mask coordinates over the canvas. Your idea sounds way better.

**Reply by GeorgLegato** [reply]
one idea results in another.  i like that
